libraries/ESP8266WiFi/src/ESP8266WiFiAP.cpp

```diff
@@ -358,25 +367,19 @@ String ESP8266WiFiAPClass::softAPmacAddress(void) {
 String ESP8266WiFiAPClass::softAPSSID() const {
     struct softap_config config;
     wifi_softap_get_config(&config);
-    char* name = reinterpret_cast<char*>(config.ssid);
-    char ssid[sizeof(config.ssid) + 1];
-    memcpy(ssid, name, sizeof(config.ssid));
-    ssid[sizeof(config.ssid)] = '\0';
-
-    return String(ssid);
+    String ssid;
+    ssid.concat(reinterpret_cast<const char*>(config.ssid), config.ssid_len);
+    return ssid;
 }
 
 /**
- * Get the configured(Not-In-Flash) softAP PSK or PASSWORD.
- * @return String psk.
+ * Get the configured(Not-In-Flash) softAP PASSWORD.
+ * @return String password.
  */
 String ESP8266WiFiAPClass::softAPPSK() const {
     struct softap_config config;
     wifi_softap_get_config(&config);
     char* pass = reinterpret_cast<char*>(config.password);
-    char psk[sizeof(config.password) + 1];
-    memcpy(psk, pass, sizeof(config.password));
-    psk[sizeof(config.password)] = '\0';
-
-    return String(psk);
+    pass[sizeof(config.password) - 1] = '\0'; // it is impossible to set PSK through the Arduino API because we only support WPA2
+    return pass;

```

Comment:
Can't the password be 64 bytes long without `\0` (like with the SSID ?)

Reply:
Not from the Arduino API as-is, pass_len > 63 fails the WiFi.softAP() call
Re-googling the WPA-Personal, my mistake here... passphrase limit is indeed 63, *but* HEX key is exactly 64